frontend/public/components/nav/nav-header.tsx

```diff
@@ -1,24 +1,88 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
-import { Title } from '@patternfly/react-core';
-import { getActivePerspective } from '../../reducers/ui';
+import { Dropdown, DropdownItem, DropdownToggle, Title } from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import {Extension} from '@console/plugin-sdk/src/typings';
 import * as plugins from '../../plugins';
 import { RootState } from '../../redux';
+import { getActivePerspective } from '../../reducers/ui';
+import * as UIActions from '../../actions/ui';
+import { history } from '../utils';
 
 type StateProps = {
   activePerspective: string;
 };
 
-export const NavHeader: React.FC<StateProps> = ({ activePerspective }) => {
-  const { icon, name } = plugins.registry
-    .getPerspectives()
-    .find((p) => p.properties.id === activePerspective).properties;
+export type NavHeaderProps = {
```

Comment:
I don't think this needs to be exported

Reply:
We should export any prop types that are part of the API of an exported component. However since `activePerspective` and `setActivePerspective` are part of the state props coming from redux connect, they shouldn't be part of the exported props.

```
type StateProps = {
  activePerspective: string;
  setActivePerspective: (id: string) => void;
}

export type NavHeaderProps = {
  onPerspectiveSelected: () => void;
}

const NavHeader_: React.FC<NavHeaderProps & StateProps> ...
```